pkg/sql/conn_executor.go

```diff
@@ -3687,3 +3759,56 @@ func init() {
 		return planGistFromCtx(ctx)
 	})
 }
+
+type jobsInfo struct {
+	jobsCreated           jobsCollection
+	uniqueJobsToCreate    map[descpb.ID]*jobs.Record
+	nonUniqueJobsToCreate []*jobs.Record
+}
+
+func newJobsInfo() *jobsInfo {
+	ret := &jobsInfo{}
+	ret.reset()
+	return ret
+}
+
+func (j *jobsInfo) addCreatedJobID(jobID ...jobspb.JobID) {
+	j.jobsCreated.add(jobID...)
+}
+
+func (j *jobsInfo) addNonUniqueJobToCreate(jobRecord *jobs.Record) {
+	j.nonUniqueJobsToCreate = append(j.nonUniqueJobsToCreate, jobRecord)
+}
+
+func (j *jobsInfo) reset() {
+	j.jobsCreated.reset()
+	j.uniqueJobsToCreate = make(map[descpb.ID]*jobs.Record)
+	j.nonUniqueJobsToCreate = nil
```

Comment:
clear the map rather than making a new one. There's a compiler intrinsic, and nobody else has the map.

```
for id := range j.uniqueJobsToCreate {
    delete(j.uniqueJobsToCreate, id)
}
```

Reply:
done.